src/ReduxCanvasV4.js

```diff
@@ -0,0 +1,61 @@
+import Immutable from 'immutable'
+import React from 'react'
+import { Provider, connect } from 'react-redux'
+import createStore from './createStore'
+import Pixel from './Pixel'
+
+// Reducer
+const store = createStore((state = Immutable.Map(), action) => {
+  if (action.type === 'TOGGLE') {
+    const key = action.i + ',' + action.j
+    return state.set(key, !state.get(key))
+  }
+  return state
+})
+
+const ACTIVE_PROPS = { active: true }
+const INACTIVE_PROPS = { active: false }
+
+// Connected pixel
+const ConnectedPixel = connect(
+  (initialState, initialProps) => {
+    const { i, j } = initialProps
+    return state => {
+      const active = state.get(i + ',' + j) || false
+      return active ? ACTIVE_PROPS : INACTIVE_PROPS
+    }
+  },
+  (initialState, initialProps) => (dispatch) => {
```

Comment:
Should the first parameter be `dispatch`? I see even though the `mapDispatch` function is a factory, in the [Redux source code](https://github.com/tgriesser/react-redux/blob/master/src/components/connect.js#L120) it would still be passed `dispatch` in the first argument. If that's the case can we do this instead

```js
(dispatch, initialProps) => () => {
  const { i, j } = initialProps
  return {
    onToggle() {
      dispatch({ type: 'TOGGLE', i, j })
    }
  };
}
```